libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/CoreBotAdapter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.AspNet.Core
+{
+    /// <summary>
+    /// CoreBotAdapter is a base BotAdapter that derives from CloudAdapter and implements OnTurnError handling. This class
+    /// also registers any middleware services that are contained in the middlewares collection.
+    /// </summary>
+#pragma warning disable CA1812 // Avoid uninstantiated internal classes. This class is only instantiated via dependency injection so no direct reference is made.
```

Comment:
Can we add unit tests for these class? (we can probably remove the #pragma once we do that). Feel free to use internals visible to like we do in [BotConnector](https://github.com/microsoft/botbuilder-dotnet/blob/a4aada9fe9ac86abb6b955516061081fb8efd0d8/libraries/Microsoft.Bot.Connector/AssemblyInfo.cs#L7)